File: src/fileUtils.ts
import * as fs from 'fs';

export function fileExistsOrNotSet(filePath: fs.PathLike) {
  return !filePath || fs.existsSync(filePath);
}

export async function loadWordListFileIfSet(filePath: fs.PathLike): Promise<string[]> {
  if (!filePath) 
    return null;

  if (!fileExistsOrNotSet(filePath)) {
    throw `${filePath} is not found`;
  }

  const words = fs.readFileSync(filePath, "utf8")
    .split("\n")
    .map(x => x.trim());

  return words;
}

File: src/file_utils.rs
use std;
use std::fs;
use std::path::Path;

pub fn fileExists(filePath: &str) -> bool {
  let exists = Path::new(filePath).exists();
  exists
}

pub fn loadWordListFileIfSet(filePath: &Option<&str>) -> Option<Vec<String>> {
  if filePath.is_none() {
    return None;
  }

  let filePath = filePath.unwrap();
  if !fileExists(&filePath) {
    panic!("file {} is not found", &filePath);
  }

  let contents = fs::read_to_string(filePath)
    .expect("Should have been able to read the file");

  let words = contents
    .split("\n")
    .map(|x| x.trim().to_owned())
    .collect::<Vec<String>>();

  Some(words)
}
